### ISSUE TITLE
🚀 Allow custom expiry per session

### ISSUE THREAD

**Issue by sujit-baniya** [gofiber/fiber#1298]
**Describe the solution you'd like**
Currently we can only define default Expiry for session. This applies for all session. I would like to be able to define Expiry at the time of login based on different parameters. Also it would be great to have following features if possible:

1) constantly update the expiration when user uses the session. Currently the session expires even though session is actively used.
2) expire session when browser is closed.
3) ability to set custom expiry

**Additional context**
Use Case:
For Remember Me context, I would like to define different Expiry Duration than the normal Expiry duration

**Comment by hi019** [comment]
For #1 we should have something like `session.SetExpiry`. For #2, that is related to https://github.com/gofiber/fiber/issues/1272

**Comment by mirusky** [comment]
Just a question, if we use something like the rate [limiter manager](https://github.com/gofiber/fiber/blob/master/middleware/limiter/manager.go#L11-L25):


```go
// limiter/manager.go
type item struct {
	exp  uint64 // This part
}
```

We could include it in [session](https://github.com/gofiber/fiber/blob/master/middleware/session/session.go#L14-L21)?

And then when we call the [save](https://github.com/gofiber/fiber/blob/master/middleware/session/session.go#L152-L155), we could do something like:


```go
if s.data.exp != 0 {
        if err := s.config.Storage.Set(s.id, s.byteBuffer.Bytes(), s.data.exp); err != nil {
                return err
        }
else {
        if err := s.config.Storage.Set(s.id, s.byteBuffer.Bytes(), s.config.Expiration); err != nil {
                return err
        }
}
```

What you think?

**Comment by mirusky** [comment]
@hi019 or @ReneWerner87  could u assing it to me? Also about the implementation what I said before make sense ?

**Comment by hi019** [comment]
@mirusky Sure. About your earlier comment, looks good but I don't completely understand the second code block. I think the default exp should be from the Config, and it can be overrided per session by doing `session.SetExpiry`

**Comment by ReneWerner87** [comment]
solved with #1363
